corehq/apps/accounting/models.py

```diff
@@ -1943,6 +1947,98 @@ def pay_invoice(self, payment_record):
         self.save()
 
 
+class CustomerInvoice(InvoiceBase):
+    # CustomerInvoice is tied to a customer level account, instead of a subscription
+    account = models.ForeignKey(BillingAccount, on_delete=models.PROTECT)
+    subscriptions = models.ManyToManyField(Subscription, default=list, blank=True)
```

Comment:
Curious what you think about adding a property instead of a new database field?  Something along the lines of

```
@property
@memoized
def subscriptions(self):
    return self.account.subscription_set()
```

Reply:
`CustomerInvoice.subscriptions` is set when we decide what subscriptions to invoice in `invoicing.py` (based on the dates of the invoicing period, `should_create_invoice`, etc), and are specific to the invoice itself. I believe `account.subscription_set()` is a list of all subscriptions that have been associated with the account, including subscriptions that we would not want to invoice

Reply:
> I believe account.subscription_set() is a list of all subscriptions that have been associated with the account, including subscriptions that we would not want to invoice

That's true, although I suppose you could add the relevant filters in this property.

How will this field be used?

Reply:
👍